src/Materials/babylon.materialHelper.ts

```diff
@@ -111,11 +111,11 @@
                     defines["DIRLIGHT" + lightIndex] = false;
 
                     var type;
-                    if (light instanceof SpotLight) {
+                    if (light.getClassName() === "SpotLight") {

```

Comment:
just wondering if string testing is a good idea regarding perd. Perhaps we could introduce a function  to return the type directly